test/Base_Tests/src/Semantic/Any_Spec.enso

```diff
@@ -38,75 +38,78 @@ add_specs suite_builder =
             Any.to_text self=Any . should_equal "Any"
             Any.to_text self=Vector . should_equal "Vector"
             Any.to_text self=[1,2,3] . should_equal "[1, 2, 3]"
-            Any.to_text Any . should_equal "Any"
-            Any.to_text Vector . should_equal "Vector"
-            Any.to_text [1,2,3] . should_equal "[1, 2, 3]"
 
             Vector.to_text . should_equal "Vector"
             Text.to_text . should_equal "Text"
             Number.to_text . should_equal "Number"
             Integer.to_text . should_equal "Integer"
 
+        group_builder.specify "Static method invocation (spec_method)" <|
             # Any.spec_method is a method that takes three arguments (self, arg1, arg2)
-            (Any.spec_method) . should_be_a Function
-            (Any.spec_method Boolean) . should_be_a Function
+            Test.expect_panic_with (Any.spec_method) No_Such_Method
+            Test.expect_panic_with (Any.spec_method Boolean) No_Such_Method
+            Test.expect_panic_with (Any.spec_method Boolean Vector) No_Such_Method
+            Test.expect_panic_with (Any.spec_method Any Boolean Vector) No_Such_Method
+            Test.expect_panic_with (Any.spec_method Date Boolean Vector) No_Such_Method
             (Any.spec_method self=Boolean) . should_be_a Function
             (Any.spec_method self=Boolean Vector) . should_be_a Function
-            Any.spec_method Boolean Vector . should_equal "Any.spec_method:{Any}{Boolean}{Vector}"
-            Any.spec_method Any Boolean Vector . should_equal "Any.spec_method:{Any}{Boolean}{Vector}"
-            Any.spec_method Date Boolean Vector . should_equal "Any.spec_method:{Date}{Boolean}{Vector}"
             Any.spec_method self=Date Boolean Vector . should_equal "Any.spec_method:{Date}{Boolean}{Vector}"
             Any.spec_method self=Any Boolean Vector . should_equal "Any.spec_method:{Any}{Boolean}{Vector}"
             Date.spec_method Boolean Vector . should_equal "Any.spec_method:{Date}{Boolean}{Vector}"
 
 
-        group_builder.specify "instance methods on different types should not have defaulted self argument" <|
+        group_builder.specify "Static method invocation (Vector)" <|
             # Vector.contains has two arguments (self elem)
-            Vector.contains . to_text . contains "Vector.type.contains" . should_be_true
-            (Vector.contains [42]) . to_text . contains "Vector.type.contains" . should_be_true
+            Test.expect_panic_with (Vector.contains) No_Such_Method
+            Test.expect_panic_with (Vector.contains [42]) No_Such_Method
+            Test.expect_panic_with (Vector.contains [42] 1) No_Such_Method
             Vector.contains self=[1,2,3] 1 . should_be_true
-            Vector.contains [1,2,3] 1 . should_be_true
 
             # Vector.is_empty has just one argument (self)
-            Vector.is_empty . to_text . contains "Vector.type.is_empty" . should_be_true
```

Comment:
Oh, no! This is bad! `Vector.is_empty` is supposed to work. Or at least we need some syntax how to get a reference to `is_empty` functions of `Vector`!

- functionally-wise one could `Vector.is_empty self=_`, but that creates a wrapper lambda function around it
- special syntax `Vector.is_empty self=` - that's a bad idea I assume

Reply:
We concluded that this should not be a problem. We will just create wrapper lambda functions, if necessary. References to instance methods should not be too frequent in our codebase anyway.